atlas/templates/Jenkinsfile.gotmpl

```diff
@@ -0,0 +1,48 @@
+pipeline {
+  agent {
+    label 'ubuntu_docker_label'
+  }
+  tools {
+    go "Go 1.11"
+  }
+  options {
+    checkoutToSubdirectory('src/github.com/{{ if .Root }}{{ .Root }}/{{ .Name }}{{ else }}{{ .Name }}{{ end }}')
+  }
+  environment {
+    GOPATH = "$WORKSPACE"
+    DIRECTORY = "src/{{ if .Root }}{{ .Root }}/{{ .Name }}{{ else }}{{ .Name }}{{ end }}"
+  }
+  stages {
+    stage("Lint") {
+      steps {
+        sh "cd $DIRECTORY && make fmt && git diff --exit-code > /dev/null"
+      }
+    }
+    stage("Test") {
+      steps {
+        sh "cd $DIRECTORY && make test"
+      }
+    }
+    stage("Build") {
+      steps {
+        withDockerRegistry([credentialsId: "<insert-the-creds-id>", url: ""]) {
+          sh "cd $DIRECTORY && make docker push"
+        }
+      }
+    }
+    stage("Push") {
+      when {
+        branch "master"
+      }
+      steps {
+        withDockerRegistry([credentialsId: "<insert-the-creds-id>", url: ""]) {
+          sh "cd $DIRECTORY && make push IMAGE_VERSION=latest"
+        }
+        withAWS(credentials:'<insert-the-creds-id>') {
+          sh "cd $DIRECTORY && make push-chart || true"
```

Comment:
I don't see push-chart rule in the makefile.gotmpl. Also, these steps should be one action or add caching to reduce duplicate work of docker image push/build. Pushing a chart requires images to be pushed and images to be built, so pushing the chart is a dependency to that target. Here's what I propose for push-chart dependencies:

make push ~> make atlas ~> build code # current
make push-chart ~> make push ~> make atlas ~> build code # needs to be added

The duplication of targets push, atlas (build code) in makefile is okay if you do caching that make can understand. The makefile.common does not do this: https://github.com/infobloxopen/atlas-cli/blob/1f82d798293377adf2f70b26ae50f001bd7049f1/atlas/templates/Makefile.common.gotmpl#L65

Every jenkins push should perform one go code build. We need caching on the atlas rule. See this file for how to do makefile image caching: https://github.com/Infoblox-CTO/athena-identity-service/blob/develop/Makefile#L98-L100

Reply:
Good  point. i'll remove the push-chart from this PR and take it as separate one.